src/bin/github-publish-release.ts

```diff
@@ -15,49 +16,57 @@ async function main() {
 
   const { user: owner, repo } = repository(pkg);
 
-  const response = await github.repos.getLatestRelease({ owner, repo });
-  if (response.status !== 200) {
-    if (response.status === 404) {
-      throw new Error(
-        `status:${response.status}, no latest release of ${repo} has been found, please do a manual release on github.`,
-      );
+  try {
+    const response = await github.repos.getLatestRelease({ owner, repo });
+    if (response.status !== 200) {
+      if (response.status === 404) {
+        throw new Error(
+          `status:${response.status}, no latest release of ${repo} has been found, please do a manual release on github.`,
+        );
+      }
+      throw new Error(`status: ${response.status}, error details: ${response.data}`);
     }
-    throw new Error(`status: ${response.status}, error details: ${response.data}`);
-  }
 
-  // eslint-disable-next-line no-console
-  console.log(`Latest release: ${response.data.name}`);
+    // eslint-disable-next-line no-console
+    console.log(`Latest release: ${response.data.name}`);
 
-  const prs = await github.pulls.list({ direction: 'desc', owner, repo, sort: 'updated', state: 'closed' });
-  if (prs.status > 200) {
-    throw new Error(`status: ${prs.status}, error details: ${prs.data}`);
-  }
+    const prs = await github.pulls.list({ direction: 'desc', owner, repo, sort: 'updated', state: 'closed' });
+    if (prs.status > 200) {
+      throw new Error(`status: ${prs.status}, error details: ${prs.data}`);
+    }
 
-  // eslint-disable-next-line lodash/prefer-lodash-method
-  const body = prs.data
-    .filter((item) => item.merged_at! > response.data.published_at!)
-    .sort((foo, bar) => bar.title.localeCompare(foo.title))
-    .map((item) => `- ${item.title} #${item.number} (by @${item.user!.login})`)
-    .join('\n');
+    // eslint-disable-next-line lodash/prefer-lodash-method
```

Comment:
Any reason you aren't using lodash here?

Reply:
from the previous refactor: i saw this block i was like nah im not touching that to save time. I'll come back for it...